source/_posts/2016-11-08-Vapor-setup.md

```diff
@@ -0,0 +1,368 @@
+
+### 4. Creating a Controller
+
+Now go back to your terminal and run ```touch Sources/App/Controllers/CarController.swift``` and again ```vapor xcode```, the Controllers folder should now contain your newly created CarController.swift
+
+Before we can make any good use of our new Controller then we have to update our ```Car.swift``` model to actually get some of Vapors awesome magic.
+
+So jump back into the ```Car.swift``` file and update the file to have the following:
+
+```swift
+//Importing the Vapor framework to make the model support dependencies
+import Vapor
+//Importing Fluent which is Vapors powering of databases and tables
+import Fluent
+
+//Subclassing our Car model from 'Model' (Vapors magic model object)
+final class Car: Model {
+    
+    //Vapor uses 'Node' as their Model ids. This is the datatype that they use to make lookup and look at the primary key in the DB's.
+    var id:Node?
+    
+    var name:String
+    var color:String
+    var milesDriven:Int
+    
+    init(name:String, color:String, milesDriven: Int) {
+        self.name = name
+        self.color = color
+        self.milesDriven = milesDriven
+    }
+    
+    //Adding one of Vapors protocols to conform to the 'Model' object. This basically makes sure that data is mapped correctly when getting extracted from a data source such as a DB.  
+    init(node: Node, in context: Context) throws {
+        id = try node.extract("id")
+        name = try node.extract("name")
+        color = try node.extract("color")
+        //We are making this underscored because that's how we normally name attributes in a database table
+        milesDriven = try node.extract("miles_driven")
+    }
+    
+    //makeNode makes sure that data can be saved into the given database (this is made super dynamic)
+    func makeNode(context: Context) throws -> Node {
+        return try Node(node: [
+            "id": id,
+            "name": name,
+            "color": color,
+            //We are making this underscored because that's how we normally name attributes in a database table
+            "miles_driven": milesDriven
+        ])
+    }
+}
+
+//This is Vapor way of making migrations, leave them empty for now. We will add it later
+extension Car: Preparation {
+    static func prepare(_ database: Database) throws {
+        //
+    }
+    
+    static func revert(_ database: Database) throws {
+        //
+    }
+}
+```
+So now our model should be ready for some cool controller action. Please now open our ```CarController.swift``` and add the following code:
+
+```swift
+import Vapor
+//Import HTTP for getting all our response types, codes, etc..
+import HTTP
+
+//Subclass our Controller from ResourceRepresentable
+final class CarController: ResourceRepresentable {
+    
+    //This will get called if the index in 'makeResource()' below will be called.
+    func index(request: Request) throws -> ResponseRepresentable {
+        
+        //Create static object
```

Comment:
This is not a static object, so just not to mislead people it would be better to remove it from the comment :D